File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/event/id/EventTechId.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.event.id;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.ObjectTechId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.event.id.EventTechId")
public class EventTechId extends ObjectTechId implements IEventId
{

    private static final long serialVersionUID = 1L;

    public EventTechId(Long techId)
    {
        super(techId);
    }

    //
    // JSON-RPC
    //

    @SuppressWarnings("unused")
    private EventTechId()
    {
        super();
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/event/id/EventTechId.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/id/ObjectTechId", "as/dto/event/id/IEventId" ], function(stjs, ObjectTechId, IEventId) {
	var EventTechId = function(techId) {
		ObjectTechId.call(this, techId);
	};
	stjs.extend(EventTechId, ObjectTechId, [ ObjectTechId, IEventId ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.event.id.EventTechId';
		constructor.serialVersionUID = 1;
	}, {});
	return EventTechId;
})